rust/kernel/lib.rs

```diff
@@ -45,6 +45,7 @@ pub mod str;
 pub mod sync;
 pub mod task;
 pub mod types;
+pub mod xarray;
```

Comment:
not urgent, but should we introduce  a `collections` module and make xarray a submod of it?